### PR TITLE
Review ex2

### PR DESCRIPTION
**Pull request by Javiercerna**

### CODE REVIEW

**Review by fedetask — COMMENTED** [review]
The code is good, I don't see any error and I like the style. It would have been better to have an object-oriented approach. In any case, good job

File: src/exc2/Visualisation.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        func_array.pop(0)
+    plt.cla()
+    plt.plot(current_array,func_array)
+    plt.xlabel ('Time')
+    plt.ylabel ('h ( t )')
+    plt.title ('h ( t ) = 3 * pi * exp(-lambda [ t ] )')
+    plt.pause(0.01)
+    go_signal = plt.get_fignums()
+
+plt.show()
```
Comment (Javiercerna) [inline src/exc2/Visualisation.py:36]: Is this necessary? By the time you exit your loop, the plot is already closed, so I think nothing should happen

File: src/exc2/Visualisation.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+current_array = []
+func_array = []
+displayed_data = 20
+
+# the time when the script started
+start_time = time.time()
+# signal to terminate the script if the plot was closed
+go_signal = True
+
+while True and go_signal:
```
Comment (Javiercerna) [inline src/exc2/Visualisation.py:18]:
```suggestion
while go_signal:
```

The `while True` part becomes redundant if you already have the `go_signal` part, right? :smile:

**Review by Javiercerna — COMMENTED** [review]
Same as Federico, the code looks good (and the "real-time" functionality is great!), but we still would like to see a OOP approach. This was written in the training material.

**Pull request closed without merging**